File: backend/src/modules/usrpreferences/usrpreferences.controller.ts
import { Controller, Get, Post, Patch, Delete, Body, Query, Param, ParseIntPipe } from '@nestjs/common';
import { ApiTags, ApiOperation } from '@nestjs/swagger';
import { UsrpreferencesService } from './usrpreferences.service';
import { PaginationDto } from '../../common/dto/pagination.dto';
import { ApiResponseDto } from '../../common/dto/api-response.dto';

@ApiTags('用户偏好设置')
@Controller('api/usrpreferences')
export class UsrpreferencesController {
  constructor(private readonly usrpreferencesService: UsrpreferencesService) {}

  @Get()
  @ApiOperation({ summary: '获取用户偏好列表' })
  async findAll(@Query() dto: PaginationDto) {
    return ApiResponseDto.success(await this.usrpreferencesService.findAll(dto));
  }

  @Get('user')
  @ApiOperation({ summary: '获取指定用户偏好' })
  async findByUser(@Query('username') username: string, @Query('domain') domain: string) {
    return ApiResponseDto.success(await this.usrpreferencesService.findByUser(username, domain));
  }

  @Post()
  async create(@Body() data: any) {
    return ApiResponseDto.success(await this.usrpreferencesService.create(data), '创建成功');
  }

  @Patch(':id')
  async update(@Param('id', ParseIntPipe) id: number, @Body() data: any) {
    return ApiResponseDto.success(await this.usrpreferencesService.update(id, data), '更新成功');
  }

  @Delete(':id')
  async remove(@Param('id', ParseIntPipe) id: number) {
    await this.usrpreferencesService.remove(id);
    return ApiResponseDto.success(null, '删除成功');
  }
}

File: kam-admin-module-user/src/main/java/io/kamailio/admin/modules/usrpreferences/controller/UsrpreferencesController.java
package io.kamailio.admin.modules.usrpreferences.controller;

import io.kamailio.admin.common.dto.ApiResponse;
import io.kamailio.admin.common.dto.PaginatedResult;
import io.kamailio.admin.common.dto.PaginationDto;
import io.kamailio.admin.modules.usrpreferences.entity.UsrPreferences;
import io.kamailio.admin.modules.usrpreferences.service.UsrpreferencesService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Tag(name = "用户偏好设置")
@RestController
@RequestMapping("/api/usrpreferences")
public class UsrpreferencesController {
    private final UsrpreferencesService service;

    public UsrpreferencesController(UsrpreferencesService service) { this.service = service; }

    @GetMapping
    @Operation(summary = "获取用户偏好列表")
    public ResponseEntity<ApiResponse<PaginatedResult<UsrPreferences>>> findAll(PaginationDto dto) {
        return ResponseEntity.ok(ApiResponse.success(service.findAll(dto)));
    }

    @GetMapping("/user")
    @Operation(summary = "获取指定用户偏好")
    public ResponseEntity<ApiResponse<List<UsrPreferences>>> findByUser(@RequestParam String username, @RequestParam String domain) {
        return ResponseEntity.ok(ApiResponse.success(service.findByUser(username, domain)));
    }

    @PostMapping
    public ResponseEntity<ApiResponse<UsrPreferences>> create(@RequestBody UsrPreferences data) {
        return ResponseEntity.ok(ApiResponse.success(service.create(data), "创建成功"));
    }

    @PatchMapping("/{id}")
    public ResponseEntity<ApiResponse<UsrPreferences>> update(@PathVariable Integer id, @RequestBody UsrPreferences data) {
        return ResponseEntity.ok(ApiResponse.success(service.update(id, data), "更新成功"));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<ApiResponse<Void>> remove(@PathVariable Integer id) {
        service.remove(id);
        return ResponseEntity.ok(ApiResponse.success(null, "删除成功"));
    }
}
